vir/simd_cvt.h

```diff
@@ -0,0 +1,56 @@
+/* SPDX-License-Identifier: LGPL-3.0-or-later */
+/* Copyright © 2023 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH
+ *                  Matthias Kretz <m.kretz@gsi.de>
+ */
+
+#ifndef VIR_SIMD_CVT_H_
+#define VIR_SIMD_CVT_H_
+
+#include "simd_concepts.h"
+#if VIR_HAVE_SIMD_CONCEPTS
+#define VIR_HAVE_SIMD_CVT 1
+
+namespace vir
+{
+  namespace detail
+  {
+    template <class From, class To>
+      struct is_simd_convertible
+      : std::false_type
+      {};
+
+    template <class T0, class A0, class T1, class A1>
+      struct is_simd_convertible<stdx::simd<T0, A0>, stdx::simd<T1, A1>>
+      : std::false_type
+      {};
+  }
+  template <typename T>
+    class cvt
+    {
+      const T& ref;
+
+    public:
+      constexpr
+      cvt(const T& x)
+      : ref(x)
+      {}
+
+      template <typename U>
```

Comment:
Just a random, stupid idea:

Why not put the `typename U` at the top like
```cpp
  template <typename U, typename T>
    class cvt
```

then `auto y = cvt<intv>(…);` would work, right? Yes, it would duplicate the static_simd_cast.

Reply:
The whole point here was to deduce `U` at the point where the conversion is needed. I.e. let the context determine `U` automagically. Yes, there are situations where `floatv(cvt(x))` seems more verbose than `cvt<floatv>(x)`, but it's actually the same number of characters. And adding another template parameter to the `cvt` type opens up so many new problems that it's just not worth it. So, no `cvt<T>(x)`, that should've been `static_cast<T>(x)` and we can't add it on top of the TS.